pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -201,7 +193,71 @@ var _ = Describe("ContainerjfrController", func() {
 				Expect(result).To(Equal(reconcile.Result{}))
 			})
 		})
+		Context("Switching from a minimal to a non-minimal deployment", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewMinimalContainerJFR(),
+				}
+			})
+			JustBeforeEach(func() {
+				reconcileFully(client, *controller, true)
+
+				cjfr := &rhjmcv1beta1.ContainerJFR{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, cjfr)
+				Expect(err).ToNot(HaveOccurred())
+
+				cjfr.Spec.Minimal = false
+				err = client.Status().Update(context.Background(), cjfr)
+				Expect(err).ToNot(HaveOccurred())
+
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				_, err = controller.Reconcile(req)
+				Expect(err).To(HaveOccurred())
+				ingressConfig(client, *controller, req, false)
+				_, err = controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			It("should create grafana resources", func() {
+				checkGrafanaService(client)
+			})
+			It("should configure deployment appropriately", func() {
+				checkDeployment(client, false)
+			})
+		})
+		Context("Switching from a non-minimal to a minimal deployment", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFR(),
+				}
+			})
+			JustBeforeEach(func() {
+				reconcileFully(client, *controller, false)
+
+				cjfr := &rhjmcv1beta1.ContainerJFR{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, cjfr)
+				Expect(err).ToNot(HaveOccurred())
+
+				cjfr.Spec.Minimal = true
+				err = client.Status().Update(context.Background(), cjfr)
+				Expect(err).ToNot(HaveOccurred())
+
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				_, err = controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			It("should delete grafana resources", func() {
+				service := &corev1.Service{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr-grafana", Namespace: "default"}, service)
+				Expect(err).To(HaveOccurred())
```

Comment:
To be sure that the error is due to the Service being deleted, you can check that it's a NotFound error with this function:
https://pkg.go.dev/k8s.io/apimachinery@v0.20.0/pkg/api/errors#IsNotFound

You can import this package as `kerrors` to avoid name collision with the standard errors package, as done elsewhere in the operator.

Reply:
Ah, this is much better, I added it to several places where I intended on checking that it wasn't found.